Loop/Managers/LoopDataManager.swift

```diff
@@ -40,6 +40,8 @@ final class LoopDataManager {
     private let logger = DiagnosticLog(category: "LoopDataManager")
 
     private let analyticsServicesManager: AnalyticsServicesManager
+    
+    private let test_currentDate: Date?
```

Comment:
The way I've done this in the past is similar to what @darinkrauss suggests above: inject a function that returns a date, and have it's default to return `Date()`. Thusly:

```
private let now: () -> Date
```

 ```
init(
    ...
        now: () -> Date = { Date() }
) {
...
    self.now = now
```

... then in all callers instead of `Date()` just call `now()` (or `self.now()`).
Then unit tests can just inject their own function.  Almost the same thing as above, just thought I'd mention it. :)

Reply:
Yes, we do this in DashPumpManager, as well.

Reply:
Done